fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializedViewOptimizer.java

```diff
@@ -28,6 +28,11 @@
 public class MaterializedViewOptimizer {
     public MvPlanContext optimize(MaterializedView mv,
                                   ConnectContext connectContext) {
+        return optimize(mv, connectContext, false);
+    }
+    public MvPlanContext optimize(MaterializedView mv,
+                                  ConnectContext connectContext,
+                                  boolean keepView) {
```

Comment:
What if the MV contains N views, can you support inline some of views and keep others ?

Reply:
now we only extract the top level view and extract it all at once. we do not support the algorithm now, but we can extend to support it later without that many modification. what has to do is that it will be more complex and add more tests then and should incur more optimization costs.